src/shared/sleep-config.c

```diff
@@ -373,13 +403,18 @@ int find_hibernate_location(HibernateLocation **ret_hibernate_location) {
                         if (r < 0)
                                 return r;
 
+                        dev_t swap_device;
+                        r = parse_dev(swap_device_id, &swap_device);
```

Comment:
First a conversion from a number to a string, and then back from a string to a number. swap_device_to_major_minor() should just return `dev_t`.

Reply:
Totally agree with @keszybz. There is no point to ask the user to set an environment variable.
When the user setup kernel parameters it is already explicit enough.

There are a few important points to point out of why this environment variable is inconvenient.

* Present: It will break existing setup. The user must figure out the new way and begs the question why the old way just work. It also means that the PR that introduces this bug is incomplete.

* Future: Eventually Systemd is going to fix this, where the memory check is done properly for  swap file on btrfs. This benefit will not propagate to the user since the environment variable is set. Once this environment variable is configured it will stay there. I am not going to monitor the changelog for when I can remove the variable to get the benefit of memory checks.

Reply:
> First a conversion from a number to a string, and then back from a string to a number. swap_device_to_major_minor() should just return `dev_t`.

Thanks for this suggestion. This was a good catch.